vignettes/MDA.Rmd

```diff
@@ -40,119 +57,263 @@ simparams <- get_parameters(
+       legend = c("MDA", "No-Int", "MDA Day"),
+       col= c(plot_cols[3:4], "black"), box.col = "white",
+       lwd = 1, lty = c(1, 1, 2), cex = 0.8)
+
+```
+
+## Seasonal Malaria Chemoprevention
+
+To demonstrate how to simulate SMC in `malariasimulation` , in the following section we'll set-up and simulate a campaign that treats 80% of children aged 3-59 months old with four doses of SP-AQ per year. We will set the SMC campaign to begin in the second year, run for two years, and specify the four SMC events to occur at monthly intervals, starting a month prior to the peak malaria season.
+
+### Interventions
+
+While the first step would typically be to establish the base set of parameters, we can use those stored earlier in `simparams`. We first store a copy of these base parameters and then use `set_drugs()` to store the preset parameters for SP-AQ in the parameter list.
+
+As SMC is conducted during the peak malaria season, we need to use the `peak_season_offset()` function to determine when the peak malaria season occurs given our specified seasonal profile. This function reads in the parameter list, uses `g0`, `g`, `h`, and `rainfall_floor` to generate the rainfall profile for a single year, and returns the day on which the maximum value occurs. We can then take this calculated seasonal peak and time SMC events around it. Here, we've specified four monthly drug administration days per year starting one month before the seasonal peak.
+
+To illustrate this, we can plot the total adult mosquito population size through time, which in the absence of interventions targeting mosquitoes will closely matches the rainfall pattern, and overlay both the annual seasonal peak and planned SMC events. Once we're happy with our SMC campaign, we use the `set_smc()` function to update the parameter list with our `drug`, `timesteps`, `coverages` and target age group (`min_ages` and `max_ages`).
+
+```{r, fig.width=7.2, fig.height=4, fig.fullwidth=TRUE}
+
+# Copy the original simulation parameters
 smcparams <- simparams
 
-# Add SMC strategy
+# Append the parameters for SP-AQ using set_drugs
+smcparams <- set_drugs(smcparams, list(SP_AQ_params))
+
+# Use the peak_season_offset() to calculate the yearly offset (in daily timesteps) for the peak mosq.
+# season
 peak <- peak_season_offset(smcparams)
-smc_events = data.frame(
-  timestep = c(1, 2) * 365 + peak - 30,
-  name=c("SMC 1", "SMC 2")
-)
+
+# Create a variable for total mosquito population through time:
+noint_output$mosq_total = noint_output$Sm_All_count + noint_output$Im_All_count + noint_output$Pm_All_count
+
+# Set-up the times (in days) around the peak to schedule drug administration days:
+admin_days <- c(-30, 0, 30, 60)
+
+# Use the peak-offset, number of simulation years and number of drug admin. days to calculate the days
+# on which to administer drugs in each year
+smc_days <- rep((365 * seq(1, years-1, by = 1)), each = length(admin_days)) + peak + rep(admin_days, 2)
```

Comment:
You probably want all of the SMC days to occur before and slightly after the peak seasonality. Right now you are giving SMC when mosquitoes are already starting their decline so it won't be as effective.

Reply:
The SMC events have been rescheduled to to occur i) 60 days before the peak, ii) 30 days before the peak, iii) on the peak day, and iv) 1 month after the peak.